core/directory.go

```diff
@@ -677,16 +677,21 @@ func (dir *Directory) Diff(ctx context.Context, other *Directory) (*Directory, e
 	return dir, nil
 }
 
-func (dir *Directory) Without(ctx context.Context, path string) (*Directory, error) {
+func (dir *Directory) Without(ctx context.Context, path string, allowWildCard bool, allowNotFound bool) (*Directory, error) {
 	dir = dir.Clone()
 
 	st, err := dir.State()
 	if err != nil {
 		return nil, err
 	}
 
+	fmt.Println("path", path)
+	fmt.Println("allowNotfound", allowNotFound)
+	fmt.Println("allowWildCard", allowWildCard)
+
+
 	path = filepath.Join(dir.Dir, path)
-	err = dir.SetState(ctx, st.File(llb.Rm(path, llb.WithAllowWildcard(true), llb.WithAllowNotFound(true))))
```

Comment:
Huh. I didn't know we defaulted to this.

This feels *slightly* inconsistent with other stuff - see another discussion on magic processing in https://github.com/dagger/dagger/pull/7173. A bit strange that some paths can be globs/wildcards, and some aren't.

Reply:
Yes definitely, I think that's why we wanted to add the parameter.